calibrators/marker_radar_lidar_calibrator/src/transformation_estimator.cpp

```diff
@@ -0,0 +1,188 @@
+  Eigen::Isometry3d calibrated_radar_to_radar_optimization_transformation(
+    full_radar_to_radar_optimization_transformation.cast<double>());
+
+  if (transformation_type == ExtrinsicReflectorBasedCalibrator::TransformationType::svd_2d) {
+    // Check that is is actually a 2D transformation
+    auto calibrated_radar_to_radar_optimization_rpy = autoware::universe_utils::getRPY(
+      tf2::toMsg(calibrated_radar_to_radar_optimization_transformation).orientation);
+    double calibrated_radar_to_radar_optimization_z =
+      calibrated_radar_to_radar_optimization_transformation.translation().z();
+    double calibrated_radar_to_radar_optimization_roll =
+      calibrated_radar_to_radar_optimization_rpy.x;
+    double calibrated_radar_to_radar_optimization_pitch =
+      calibrated_radar_to_radar_optimization_rpy.y;
+
+    if (
+      calibrated_radar_to_radar_optimization_z != 0.0 ||
+      calibrated_radar_to_radar_optimization_roll != 0.0 ||
+      calibrated_radar_to_radar_optimization_pitch != 0.0) {
+      RCLCPP_ERROR(
+        rclcpp::get_logger("marker_radar_lidar_calibrator"),
+        "The estimated 2D translation was not really 2D. Continue at your own risk. z=%.3f "
+        "roll=%.3f "
+        "pitch=%.3f",
+        calibrated_radar_to_radar_optimization_z, calibrated_radar_to_radar_optimization_roll,
+        calibrated_radar_to_radar_optimization_pitch);
+    }
+
+    calibrated_radar_to_radar_optimization_transformation.translation().z() =
+      (initial_radar_to_lidar_eigen_ * radar_optimization_to_lidar_eigen_.inverse())
+        .translation()
+        .z();
+  }
+
+  calibrated_radar_to_lidar_transformation_ =
+    calibrated_radar_to_radar_optimization_transformation * radar_optimization_to_lidar_eigen_;
+}
+
+void TransformationEstimator::estimateZeroRollTransformation()
+{
+  RCLCPP_INFO(
+    rclcpp::get_logger("marker_radar_lidar_calibrator"),
+    "Estimate 3D transformation with roll is always zero");
+
+  ceres::Problem problem;
+
+  Eigen::Vector3d translation = initial_radar_optimization_to_radar_eigen_.translation();
+  Eigen::Matrix3d rotation = initial_radar_optimization_to_radar_eigen_.rotation();
+  Eigen::Vector3d euler_angle = rotation.eulerAngles(0, 1, 2);
+
+  // params: x, y, z, pitch, yaw
+  std::vector<double> params = {
+    translation[0], translation[1], translation[2], euler_angle[1], euler_angle[2]};
+
+  std::string initial_params_str = std::accumulate(
+    std::next(params.begin()), params.end(),
+    std::to_string(params.front()),  // Initialize with the first element
+    [](const std::string & a, const auto & b) { return a + " " + std::to_string(b); });
+  std::string initial_params_msg = "initial params (x,y,z,pith,yaw): " + initial_params_str;
+  RCLCPP_INFO(
+    rclcpp::get_logger("marker_radar_lidar_calibrator"), "%s", initial_params_msg.c_str());
+
+  for (size_t i = 0; i < lidar_points_ocs_->points.size(); i++) {
```

Comment:
We have been trying to use std::size_t so far

Reply:
addressed in https://github.com/tier4/CalibrationTools/pull/180/commits/28fdcf92c92194da78cae2678541c7a4a1e62b92